src/Codeception/Test/Unit.php

```diff
@@ -63,32 +58,37 @@ public function setMetadata(?Metadata $metadata): void
 
     public function getResultAggregator(): ResultAggregator
     {
-        throw new LogicException('This method should not be called, TestCaseWrapper class must be used instead');
+        throw new LogicException('This method should not be called; use TestCaseWrapper instead.');
     }
 
     protected function _setUp()
     {
-        if ($this->getMetadata()->isBlocked()) {
-            if ($this->getMetadata()->getSkip() !== null) {
-                $this->markTestSkipped($this->getMetadata()->getSkip());
+        $metadata = $this->getMetadata();
+        if ($metadata->isBlocked()) {
+            if ($skip = $metadata->getSkip()) {
+                $this->markTestSkipped($skip);
             }
-            if ($this->getMetadata()->getIncomplete() !== null) {
-                $this->markTestIncomplete($this->getMetadata()->getIncomplete());
+            if ($incomplete = $metadata->getIncomplete()) {
+                $this->markTestIncomplete($incomplete);
             }
             return;
         }
 
         /** @var Di $di */
-        $di = $this->getMetadata()->getService('di');
-        // auto-inject $tester property
-        if (($this->getMetadata()->getCurrent('actor')) && ($property = lcfirst((string) Configuration::config()['actor_suffix']))) {
-            $this->$property = $di->instantiate($this->getMetadata()->getCurrent('actor'));
+        $di = $metadata->getService('di');
+
+        // Auto-inject $tester property
+        if ($actor = $metadata->getCurrent('actor')) {
+            $suffix   = (string) Configuration::config()['actor_suffix'];
+            $property = lcfirst($suffix);
+            if (property_exists($this, $property)) {
+                $this->{$property} = $di->instantiate($actor);
+            }
```

Comment:
This check breaks some tests in one of my own projects if I try to access `$this->tester`. Error message: ` Undefined property: xxxTest::$tester`

@TavoNiievez Should this be `!property_exists($this, $property)` instead?

Reply:
@W0rma Sounds like you're right. You can open a PR with something like this:
```php
$di = $this->getMetadata()->getService('di');

if (($actor = $this->getMetadata()->getCurrent('actor')) &&
    ($property = lcfirst((string) Configuration::config()['actor_suffix']))) {

    $this->{$property} = $di->instantiate($actor);
}
```
?

Reply:
@W0rma Sounds like you're right. Could you open a PR with something like this:
```php
$di = $this->getMetadata()->getService('di');

if (($actor = $this->getMetadata()->getCurrent('actor')) &&
    ($property = lcfirst((string) Configuration::config()['actor_suffix']))) {

    $this->{$property} = $di->instantiate($actor);
}
```
?

Reply:
@TavoNiievez See https://github.com/Codeception/Codeception/pull/6856

The change fixes the tests of my own project.